### PR TITLE
Add several tips to Lambda, add additional gotchas to Lambda, rewrite…

### PR DESCRIPTION
**Pull request by andre-meireles**
… Lambda basics statement, fix incorrect tmp storage limit in Lambda

Hi folks,

This is my first github PR! 

My approach to editing the Lambda section was to pull the focus a bit tighter around the Lambda service itself, although I retained the majority of the content mentioning serverless as a concept. 

On further reflection, I think that this guide could also benefit from a section with higher-level usage patterns on AWS. Some sections we could write include serverless architecture (Lambda, APIG, Batch), data warehousing and analytics (Redshift, Glue, Kinesis, etc), and machine learning and AI (Lex, Polly, Rekognition, etc), among others. Thoughts on this?

Andre

Full disclosure: I am an employee of Amazon but this PR is in no away affiliated with my employer and does not reflect my employer's views. I am not representing my employer in any capacity.

### CODE REVIEW

**Comment by Ashex** [comment]
Great job with the refresh! I'd like to ask that you add a little more on how Lambda handles failed invocations, in particular how Lambda will automatically retry with asynchronous invocations. 

http://docs.aws.amazon.com/lambda/latest/dg/retries-on-errors.html

This is one of those gotchas that people typically aren't aware of and spend a lot of time digging into.

**Comment by QuinnyPig** [comment]
It... it retires on failure?

Oh my god.

**Comment by QuinnyPig** [comment]
@andre-meireles Thanks for this. Your PR is a thing of beauty, and sorely needed. Are you in the Slack team?

File: README.md [hunk 1]
```diff
@@ -1351,32 +1351,43 @@ Lambda
...
--	The [Awesome Serverless](https://github.com/anaibol/awesome-serverless) list gives a good set of examples of the relatively new set of tools and frameworks around Lambda.
--	The [**Serverless framework**](https://github.com/serverless/serverless) is a leading new approach designed to help group and manage Lambda functions. It’s approaching version 1 as of August 2016) and is popular among a small number of users.
+-	The idea behind 'serverless' is that users don't manage provisioning, scaling, or maintenance of the physical machines that host their application code. With Lambda, the machine that actually executes the user-defined function is abstracted as a ['container'](http://docs.aws.amazon.com/lambda/latest/dg/lambda-introduction.html). When defining a Lambda function, users are able to declare the amount of memory available to the function, which directly affects the physical hardware specification of the Lambda container.
+-	Changing the amount of memory available to your Lambda functions also affects the amount of [CPU power](https://aws.amazon.com/lambda/faqs/) available to it.
```
Comment (ryansb) [inline README.md]: Network bandwidth also scales with memory.

**Pull request merged**